cordyceps/src/list.rs

```diff
@@ -897,6 +913,17 @@ impl<T: Linked<Links<T>> + ?Sized> List<T> {
         }
     }
 
+    /// Returns an iterator over the items in this list, by pointer.
```

Comment:
i don't think that the act of _constructing_ a `RawIter` should be unsafe --- it's analogous to constructing a raw pointer being safe, but dereferencing it being unsafe. however, I _do_ think the docs for this method should discuss the safety invariants for using the raw pointers handed out by the `RawIter`. in particular, you definitely cannot move out of them --- they are "logically pinned" even though they cannot be `Pin<NonNull<T>>`[^1] --- and we ought to state that here. 

[^1]: see https://github.com/rust-lang/rust/issues/54815